R/loadFromMetaphlan.R

```diff
@@ -0,0 +1,103 @@
+#' Import Metaphlan results to \code{SummarizedExperiment}
+#
+#' @param file a single \code{character} value defining the file
+#'   path of the metaphlan file.
+#'
+#' @param ... additional arguments:
+#' \itemize{
+#'   \item{\code{removeTaxaPrefixes}:} {\code{TRUE} or \code{FALSE}: Should
+#'     taxonomic prefixes be removed? (default:
+#'     \code{removeTaxaPrefixes = FALSE})}
+#' }
+#'
+#' @details
+#' Import metaphlan results
+#'
+#' @return  A
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object
+#'
+#' @name loadFromMetaphlan
+#' @seealso
+#' \code{\link[=makeTreeSummarizedExperimentFromPhyloseq]{makeTreeSummarizedExperimentFromPhyloseq}}
+#' \code{\link[=makeSummarizedExperimentFromBiom]{makeSummarizedExperimentFromBiom}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=loadFromQIIME2]{loadFromQIIME2}}
+#' \code{\link[=loadFromMothur]{loadFromMothur}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#' 
+#' @references
+#'
+#' @examples
+#' \dontrun{
+#' # File path
+#' file_path <- "/data/metaphlan_result.txt"
+#' # Import data
+#' se <- loadFromMetaphlan(file)
+#' se
+#' }
+#' 
+
+loadFromMetaphlan <- function(file, ...){
+    ################################ Input check ################################
+    if(!.is_non_empty_string(file)){
+        stop("'file' must be a single character value.",
+             call. = FALSE)
+    }
+    if (!file.exists(file)) {
+        stop(file, " does not exist", call. = FALSE)
+    }
+    ############################## Input check end #############################
+    # Read the table
+    table <- read.table(file, header = TRUE)
+    # Subset so that only those rows are included that include all taxonomic levels
+    table <- .get_rows_that_include_lowest_level(table)
+    # Get the data that belongs to rowData
+    rowdata_columns <- c("clade_name", "NCBI_tax_id")
```

Comment:
Better to include arguments for these in function. The names can be variable. Is this for version metaphlan3 or 2 ?
![image](https://user-images.githubusercontent.com/16476233/142611296-a43e24dd-af9e-45d0-ae68-9a468578897a.png)
Or position based selection cols numbers one and two.